### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by AlaaAlsalem**
Here are the things I added to this repo
>Add RSpec as a gem
>Create a Request spec file for all controllers.

### CODE REVIEW

File: test/controllers/posts_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  test 'should get index' do
+    get posts_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get posts_show_url
+    assert_response :success
+  end
+end
```
Comment (Deepakdanger) [inline test/controllers/posts_controller_test.rb]: - [ ] These files should be under `spec/requests` directory to run the test not inside the `test` file.
- [ ] Here, I think you should use `type: request` as I have shown in the example code.

Example:

```ruby
require 'rails_helper'

 RSpec.describe "Posts", type: :request do
   ...
 end
```

- [ ] Inside this there should be two describe blocks one for `index` and another for `show` where you will test if the index or show template is rendering, if it includes the text you expect on the page, etc..

Hint:


```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

File: test/controllers/users_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  test 'should get index' do
+    get users_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get users_show_url
+    assert_response :success
+  end
+end
```
Comment (Deepakdanger) [inline test/controllers/users_controller_test.rb]: - [ ] Same for User specs what I have mentioned the change in posts spec.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @AlaaAlsalem, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

- [ ] I would recommend including the command to test the specs you have written in the README file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/controllers/users_controller_test.rb (hunk 2 above)
Comment (AlaaAlsalem) [reply test/controllers/users_controller_test.rb]: Hello @Deepakdanger , thanks for your feedback , they are all resolved

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @AlaaAlsalem  😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the changes requested by the previous reviewer 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**